Userland/Utilities/grep.cpp

```diff
@@ -186,15 +187,19 @@ int main(int argc, char** argv)
     };
 
     if (!files.size() && !recursive) {
-        auto stdin_file = Core::File::standard_input();
-        while (!stdin_file->eof()) {
-            auto line = stdin_file->read_line();
```

Comment:
I think the same bug exists in `Core::File::LineIterator` as well then (which I hope replaces the manual way of reading lines from a `Core::File` eventually)

seems like `IODevice::read_line()` should just block until it gets a newline instead of returning an empty string when it doesn't

Reply:
It kinda feels weird that there's two completely separate implementations of buffered file I/O in the system. FILE* seems to be the more fleshed out one, at least recently. Shame it's got a crusty C API.

Reply:
Yeah, I've tried making `IODevice` use a `FILE*` before, but it got rather boring to me so I gave up...

(just to clarify, I'm not saying these should be done in this PR)

Reply:
Yeah it feels like the Core::File class works great for like, actual files that have a known beginning and end. When we start talking about streams it becomes less convincing.

If other parts of the system are actually using Core::File::standard_input (or Core::File::construct(STDIN_FILENO, ...) because the author didn't read the entire header file) it's likely they'll get surprising empty strings instead of full lines when using read_line.

Reply:
I wrote up https://github.com/SerenityOS/serenity/issues/5093 to discuss this more. Seems like a bigger thing than grep -v for sure :)